app/policies/workflow_policy.rb

```diff
@@ -0,0 +1,25 @@
+class WorkflowPolicy < ApplicationPolicy
+  delegate :editor?, :approver?, :viewer?, to: :project_policy
+
+  def project_policy
+    ProjectPolicy.new(user, projects)
+  end
+
+  def projects
+    records.collect(&:project).uniq
+  end
+
+  class Scope < Scope
+    def resolve
+      viewer_policy_scope
+    end
+
+    def viewer_policy_scope
+      if user.admin?
+        scope.all
+      elsif user
+        scope.joins(:project).where project_id: viewer_project_ids
```

Comment:
No need to do the join here, workflow table has the project_id on it.
```suggestion
        scope.where project_id: viewer_project_ids
```

Reply:
Ah, thanks--this was a vestige of when I was sharing that method from the app policy, but they all needed to be different, turns out.